neuraxle/metaopt/random.py

```diff
@@ -114,6 +117,103 @@ def _split(self, data_inputs):
+        :param data_inputs: data to perform walk forward cross validation into.
+        :param expected_outputs: the expected target*label that will be used during walk forward cross validation.
+        :return: train_data_inputs, train_expected_outputs, validation_data_inputs, validation_expected_outputs
+        """
+        # TODO: Verify that we need to split into training and validation or into inputs and expected_outputs.
+        validation_data_inputs, validation_expected_outputs = self.validation_split(
+            data_inputs, expected_outputs)
+
+        train_data_inputs, train_expected_outputs = self.train_split(
+            data_inputs, expected_outputs)
+
+        return train_data_inputs, train_expected_outputs, validation_data_inputs, validation_expected_outputs
+
+    def train_split(self, data_inputs, expected_outputs=None) -> (List, List):
+        """
+        Split the data into train inputs, train expected outputs
+        :param data_inputs: data to perform walk forward cross validation into.
+        :param expected_outputs: the expected target*label that will be used during walk forward cross validation.
+        :return: train_data_inputs, train_expected_outputs
+        """
+        splitted_data_inputs = self._train_split(data_inputs)
+        if expected_outputs is not None:
+            splitted_expected_outputs = self._train_split(expected_outputs)
+            return splitted_data_inputs, splitted_expected_outputs
+
+    def validation_split(self, data_inputs, expected_outputs=None) -> List:
+        """
+        Split the data into validation inputs, valiadtion expected outputs
+        :param data_inputs: data to perform walk forward cross validation into.
+        :param expected_outputs: the expected target*label that will be used during walk forward cross validation.
+        :return: validation_data_inputs, validation_expected_outputs
+        """
+        splitted_data_inputs = self._validation_split(data_inputs)
+        if expected_outputs is not None:
+            splitted_expected_outputs = self._validation_split(expected_outputs)
+            return splitted_data_inputs, splitted_expected_outputs
+        return splitted_data_inputs
+
+    def _train_split(self, data_inputs):
+        splitted_data_inputs = []
+        validation_start = self.initial_window_number * self.window_size + self.window_delay_size
+        number_step = math.ceil((data_inputs.shape[1] - validation_start) / float(self.window_size))
+
+        for i in range(number_step):
+            # first slice index is always 0 for walk forward cross validation
+            a = 0
+            b = int((self.initial_window_number + i) * self.window_size)
+
+            if b > data_inputs.shape[1]:
+                b = data_inputs.shape[1]
+
+            # TODO: Do we need to add a slicer. Does the data always come as numpy array with time series in axis=1.
+            slice = data_inputs[:, a:b]
```

Comment:
Good question. To me it seems ok for this PR, although at some point it'd be good to have such a BaseStep, and so that it be able to handle both numpy arrays and regular lists of lists.